### ISSUE TITLE
Unexpected missing top-level module documentation comment

### ISSUE THREAD

**Issue by fabiopelosin** [bbatsov/rubocop#947]
This report is relative to the `Documentation` cop. The output of Rubocop is the following:

```console
$ rubocop --version
0.20.0
$ rubocop
Inspecting 38 files
..C................C...CCC.....C......

Offenses:

lib/cocoapods-core/gem_version.rb:1:1: C: Missing top-level module documentation comment.
module Pod
^^^^^^
[...]
```

The module Pod is commented [here](https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core.rb#L1-L3) and the file which lead to the warning is this [one](https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/gem_version.rb).

Is this the intended behaviour?

**Comment by dmolesUC3** [comment]
It would be nice to separate the class and module documentation checks, at least.

**Comment by segiddins** [comment]
Might it make sense to allow namespaces that have only constant children of any kind (Class, Module, Constant) to be undocumented?

**Comment by alexdowad** [comment]
> Might it make sense to allow namespaces that have only constant children of any kind (Class, Module, Constant) to be undocumented?

This is very easy to implement. But do the maintainers want it? @bbatsov @jonas054

**Comment by jonas054** [comment]
Slightly expanding what we mean by *namespace* module. I'd be ok with that.

**Comment by alexdowad** [comment]
OK, sounds like @jonas054 likes the idea, at least a little bit. Almost all the specs for this cop will have to be adjusted. But that's OK.

**Comment by alexdowad** [comment]
Pushed a fix to my open PR. @bbatsov, if you decide that classes/modules with only constants are not "namespaces", I will delete that commit from the PR.
